jupyter_server_proxy/config.py

```diff
@@ -162,3 +162,25 @@ class ServerProxy(Configurable):
         """,
         config=True
     )
+
+    host_whitelist_hook = Any(
```

Comment:
I think this should instead be a Union of a List or Callable, and be called 'host_whitelist'. So you can either specify a list (which can then also be passed in via the commandline on starting notebook), or a callable.

There's an example of how to do that in https://github.com/yuvipanda/nbresuse/blob/master/nbresuse/__init__.py#L136.

Reply:
Agreed.

Reply:
An issue is traitlets 4.3.3 doesn't include `Callable()` and it seems to be the most recent release thought it's there in master.

Reply:
@rcthomas yeah, unfortunately you need to copy the implementation. That's what nbresuse does

Reply:
This should be in now thanks to @rcthomas